src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java

```diff
@@ -78,6 +81,39 @@ public void createPrimaryUserAffiliation(String eventPayload) {
     }
   }
 
+  @Override
+  @SneakyThrows
+  @Transactional
+  public void updatePrimaryUserAffiliation(String eventPayload) {
+    try {
+      var userEvent = objectMapper.readValue(eventPayload, UserEvent.class);
+      log.info("Received event for update primary affiliation for user: {} and tenant: {}", userEvent.getUserDto().getId(), userEvent.getTenantId());
+
+      var consortiaTenant = tenantService.getByTenantId(userEvent.getTenantId());
+      if (consortiaTenant == null) {
+        log.warn(TENANT_NOT_EXISTS_IN_CONSORTIA, userEvent.getTenantId());
+        return;
+      }
+      UUID userId = getUserId(userEvent);
+      String newUsername = userEvent.getUserDto().getUsername();
+
+      UserTenantEntity userTenant = userTenantService.getByUserIdAndTenantId(userId, userEvent.getTenantId());
+      boolean isUsernameChanged = ObjectUtils.notEqual(userTenant.getUsername(), newUsername);
+
+      if (isUsernameChanged) {
+        userTenantService.updateUsernameInPrimaryUserTenantAffiliation(userId, newUsername, userEvent.getTenantId());
+        log.info("Username in primary affiliation has been updated for the user: {}", userEvent.getUserDto().getId());
+      }
+
+      PrimaryAffiliationEvent affiliationEvent = createPrimaryAffiliationEvent(userEvent);
```

Comment:
PrimaryAffiliationEvent should also include email, phoneNumber, mobilePhoneNumber, centralTenantId

Reply:
added